### DISCUSSION TITLE
Response Transformer and JSON array

### DISCUSSION THREAD

**Discussion by waltari2001** [Kong/kong#7287] [Ideas and feature requests]
### Summary

Hi,

I am trying to add keys to the JSON body returned to me by the upstream_url.  (sample below)
```
[
  {
    "metadata": {
      "record_type": "S",
      "zip_type": "Standard",
      "dst": true
    },
    "analysis": {
      "dpv_footnotes": "AABB",
      "dpv_vacant": "N"
    }
  }
]
```
I have applied 
```
curl -X POST http://localhost:8001/consumers/consumer1/plugins \
   --data "name=response-transformer" --data "config.add.json=status:200,timestamp:2019-01-01"
```
Which results in 
```
{
  "1": {
    "analysis": {
      "dpv_footnotes": "AABB",
      "dpv_vacant": "N"
    },
    "metadata": {
      "record_type": "S",
      "zip_type": "Standard",
      "dst": true
    }
  },
  "status": "200",
  "timestamp": "2019-01-01"
}
```
Not sure what the 1 represents (array index?) but I would like to rename it to something else:

```
{
  "response": {
    "analysis": {
      "dpv_footnotes": "AABB",
      "dpv_vacant": "N"
    },
    "metadata": {
      "record_type": "S",
      "zip_type": "Standard",
      "dst": true
    }
  },
  "status": "200",
  "timestamp": "2019-01-01"
}
```

Any way that could be done?

Thanks !

### Additional Details & Logs

- Kong version (`0.14.1)

**Comment by Kerl1310** [comment]
So this looks to me like it's putting the chunk number into the response it's generating in the Transform handler... I'll take a further look at this and hopefully make a PR

**Comment by hishamhm** [comment]
@waltari2001 this is a known limitation of the plugin: it acts by changing keys in the top-level object returned; given that the top-level object is an array, adding string-key entries to it ends up turning it into an object. Given that we don't have a `rename` operation (to change the key and keep the value), the plugin cannot produce the result shape you wanted, indeed. In that sense, the behavior you requsted is more of a feature request than a bug. :)

We're not working on this feature right now but we definitely accept PRs if you'd like to try your hand at it! Thank you for the feedback!!
